src/mds/mdstypes.cc

```diff
@@ -542,8 +542,8 @@ void session_info_t::dump(Formatter *f) const
   f->close_section();
 
   f->open_array_section("used_inos");
-  for (interval_set<inodeno_t>::const_iterator p = prealloc_inos.begin();
-       p != prealloc_inos.end();
+  for (interval_set<inodeno_t>::const_iterator p = used_inos.begin();
+       p != used_inos.end();
```

Comment:
FYI Jeff, in general if you're already updating old code, you can go ahead and change the loops with iterators to use the generic form:

`for (const auto& foo : used_inos)`

Which is broadly applicable unless you're doing something like deleting elements or doing funky things with iterators.

Reply:
Done. Also switched the prealloc_inos loop to use a generic iterator.

Reply:
Hmm, that didn't seem to work. It built for me locally somehow, but in jenkins the compiler seemed to be confused about the type:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mds/mdstypes.cc:536:33: error: 'const struct std::pair<inodeno_t, inodeno_t>' has no member named 'get_start'
     f->dump_unsigned("start", p.get_start());
                                 ^~~~~~~~~
```
I'm going to suggest we don't make that change in this set.

Reply:
Ah, it needs access to the interval_set iterators. nevermind then :)